packages/phone-number-privacy/combiner/src/match-making/get-contact-matches.ts

```diff
@@ -36,16 +42,37 @@ export async function handleGetContactMatches(
       return
     }
 
-    const { account, userPhoneNumber, contactPhoneNumbers, hashedPhoneNumber } = request.body
+    const {
+      account,
+      userPhoneNumber,
+      contactPhoneNumbers,
+      hashedPhoneNumber,
+      blindedQueryPhoneNumber,
+    } = request.body
 
     if (!(await isVerified(account, hashedPhoneNumber, getContractKit(), logger))) {
       respondWithError(response, 403, WarningMessage.UNVERIFIED_USER_ATTEMPT_TO_MATCHMAKE, logger)
       return
     }
+
     if (await getDidMatchmaking(account, logger)) {
-      respondWithError(response, 403, WarningMessage.DUPLICATE_REQUEST_TO_MATCHMAKE, logger)
-      return
+      const blindedPhoneNumberRecord = await getAccountBlindedPhoneNumber(account, logger)
+      if (blindedPhoneNumberRecord !== blindedQueryPhoneNumber) {
+        if (blindedPhoneNumberRecord === 'empty') {
```

Comment:
Can you add a small comment here stating that this is for backward compatibility?